RecoLocalMuon/GEMSegment/plugins/ME0SegmentAlgorithm.cc

```diff
@@ -257,60 +259,56 @@ ME0SegmentAlgorithm::chainHits(const ME0Ensemble& ensemble, const EnsembleHitCon
   return rechits_chains;
 }
 
-bool ME0SegmentAlgorithm::isGoodToMerge(const ME0Ensemble& ensemble, const EnsembleHitContainer& newChain, const EnsembleHitContainer& oldChain) {
+bool ME0SegmentAlgorithm::isGoodToMerge(const ME0Chamber * chamber, const HitAndPositionPtrContainer& newChain, const HitAndPositionPtrContainer& oldChain) {
   for(size_t iRH_new = 0;iRH_new<newChain.size();++iRH_new){
-    GlobalPoint pos_new = ensemble.first->toGlobal(newChain[iRH_new]->localPosition());
-          
+    GlobalPoint pos_new = newChain[iRH_new]->gp;
+
     for(size_t iRH_old = 0;iRH_old<oldChain.size();++iRH_old){
-      GlobalPoint pos_old = ensemble.first->toGlobal(oldChain[iRH_old]->localPosition());
+      GlobalPoint pos_old = oldChain[iRH_old]->gp;
       // to be chained, two hits need to be in neighbouring layers...
       // or better allow few missing layers (upto 3 to avoid inefficiencies);
       // however we'll not make an angle correction because it
-      // worsen the situation in some of the "regular" cases 
+      // worsen the situation in some of the "regular" cases
       // (not making the correction means that the conditions for
       // forming a cluster are different if we have missing layers -
-      // this could affect events at the boundaries ) 
+      // this could affect events at the boundaries )
 
       // to be chained, two hits need also to be "close" in phi and eta
       if (std::abs(reco::deltaPhi( float(pos_new.phi()), float(pos_old.phi()) )) >= dPhiChainBoxMax) continue;
       if (std::abs(pos_new.eta()-pos_old.eta()) >= dEtaChainBoxMax) continue;
       // and the difference in layer index should be < (nlayers-1)
-      if (std::abs(newChain[iRH_new]->me0Id().layer() - oldChain[iRH_old]->me0Id().layer()) >= (ensemble.first->id().nlayers()-1)) continue;
-      // and they should have a time difference compatible with the hypothesis 
+      if (std::abs(newChain[iRH_new]->layer - oldChain[iRH_old]->layer) >= (chamber->id().nlayers()-1)) continue;
```

Comment:
std::abs of the difference between two unsigned int's: that difference cannot be negative.
(See also https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-17687/18303/buildClang.log)

Quick test:

unsigned int a=7;
unsigned int b=9;
std:: cout << " a = " << a << " ; b = " << b << " ; a - b = " << a-b << " ; abs = " << std::abs(a-b) << std::endl;

gives

 a = 7 ; b = 9 ; a - b = 4294967294 ; abs = 4.29497e+09